### ISSUE TITLE
no-dupe-keys: false negative between a normal property and a getter.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6801]
**What version of ESLint are you using?**

- 3.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

    $ echo "var obj = {a:1, get a() { return 1 }}" | eslint --stdin --rule no-dupe-keys:2 --env es6 --no-eslintrc

**What did you expect to happen?**

- Duplicate key 'a'. (no-dupe-keys)

**What actually happened? Please include the actual, raw output from ESLint.**

- No errors.

**Comment by btmills** [comment]
This is interesting:

```sh
$ git log --oneline -n 1
5d063f2 Docs: Add plugin example to disabling with comments guide (fixes #6742)

$ rm -rf node_modules && npm install
# ...

$ echo "var obj = {a:1, get a() { return 1 }}" | bin/eslint.js --stdin --rule no-dupe-keys:2 --env es6 --no-eslintrc
```

But if I remove `--env es6`:

```sh
$ echo "var obj = {a:1, get a() { return 1 }}" | bin/eslint.js --stdin --rule no-dupe-keys:2 --no-eslintrc

<text>
  1:21  error  Parsing error: Redefinition of property

✖ 1 problem (1 error, 0 warnings)
```

**Comment by platinumazure** [comment]
@btmills `--env es6` both enables an environment and changes a `parserOptions` property (`ecmaVersion: 6`). Can you please use `--parserOptions ecmaVersion:6` in your tests to help isolate env vs parserOptions?

**Comment by btmills** [comment]
Good call, @platinumazure. I get no errors:

```sh
$ echo "var obj = {a:1, get a() { return 1 }}" | bin/eslint.js --stdin --rule no-dupe-keys:2 --parser-options ecmaVersion:6 --no-eslintrc

```

**Comment by mysticatea** [comment]
Yes. Duplicate properties were a syntax error until ES5. But ES6 relaxed this restriction due to computed properties.
